features/dd-sdk-android-trace/src/test/kotlin/com/datadog/opentracing/SpanCleanerTest.kt

```diff
@@ -0,0 +1,184 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.opentracing
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.opentracing.PendingTrace.SpanCleaner
+import com.datadog.tools.unit.createInstance
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.quality.Strictness
+import java.math.BigInteger
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(value = Configurator::class)
+internal class SpanCleanerTest {
+
+    private lateinit var testeCleaner: SpanCleaner
+
+    @BeforeEach
+    fun setUp() {
+        testeCleaner = SpanCleaner()
+    }
```

Comment:
```suggestion
    private val testedCleaner: SpanCleaner = SpanCleaner()
```

Reply:
Hmm not sure I want that, I want to start fresh for each test.

Reply:
it will start fresh for each test - each test will have its own enclosing instance of the test class